### PR TITLE
Oauth integration

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

This PR delivers the support for OAuth authentication to Strimzi. It adds:
* Server side OAuth support to Kafka Brokers
* Client side OAuth support to Bridge, Mirror Maker and Connect
* As part of the client side OAuth I also refactored the the way authentication is done in the different compoenents (Bridge, Mirror Maker and Connect). I removed the compoenent specific authentication classes in `api` module and replaced them with one set of clases. I also added a new class `ClientAuthenticationUtils` to `cluster-operator` module which has most logic for configuring these from volumes to environment variables. This refactoring allows us to implement authentication only once and not 3 times.

This PR is still missing tests. It also depends on the not-merged OAuth callback library (strimzi/strimzi-kafka-oauth#1). Part of the Bridge implementation (the configuration and startup scripts) are in strimzi/strimzi-kafka-bridge#336. 

Additional refactoring of the environment variables used to configure the client authentication in different compoenents (currently each has its own names) would be desired. But I decided to not go for it in this PR because of too big risk of screwing up somehting when renaming the env vars in several shell scripts.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

File: docker-images/kafka/scripts/kafka_mirror_maker_tls_prepare_certificates.sh [hunk 1]
```diff
@@ -26,6 +26,8 @@ tls_auth_key=$3
 truststore_path=$4
 keystore_path=$5
 certs_key_path=$6
+oauth_certs_path=$6
+oauth_keystore_path=$6
```
Comment (ppatierno) [inline docker-images/kafka/scripts/kafka_mirror_maker_tls_prepare_certificates.sh]: both $6 ?

File: api/src/main/java/io/strimzi/api/kafka/model/GenericSecretSource.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a generic secret inside a Kubernetes Secret
```
[the remark this answers is not in the document]
Comment (scholzj) [reply api/src/main/java/io/strimzi/api/kafka/model/GenericSecretSource.java]: It is, but I do no have any better term. Actualy what I store there is a _OAuth client secret_ :-(

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java [hunk 3]
```diff
@@ -67,6 +63,7 @@
     protected static final String CO_ENV_VAR_CUSTOM_ANNOTATIONS = "STRIMZI_CUSTOM_KAFKA_BRIDGE_SERVICE_ANNOTATIONS";
 
     // Kafka Bridge configuration keys (EnvVariables)
+    protected static final String ENV_VAR_PREFIX = "KAFKA_BRIDGE_";
```
[the remark this answers is not in the document]
Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java]: Honestly, I think we should do a different thing ... we should instead stop using htese prefixes for things which are everywhere and have these environment variables whihc are in 3 different cluster the same. But I didn't wanted to make the PR even more big :-o.

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by scholzj** [comment]
I added tests now. So the only thing missing is to release the OAuth plugin and update the dependency version.

**Comment by scholzj** [comment]
@strimzi-ci run tests
testcase=io.strimzi.systemtest.RecoveryST#io.strimzi.systemtest.RecoveryST

**Comment by scholzj** [comment]
@strimzi-ci run tests
testcase=io.strimzi.systemtest.RecoveryST#io.strimzi.systemtest.RecoveryST

**Pull request merged**